rules/ast/ast.go

```diff
@@ -371,10 +383,15 @@ func (node *ScalarFunctionCall) Eval(timestamp clientmodel.Timestamp) clientmode
 	return node.function.callFn(timestamp, node.args).(clientmodel.SampleValue)
 }
 
-func (node *VectorAggregation) labelsToGroupingKey(labels clientmodel.Metric) uint64 {
+// hashForLabels returns a hash value taken from the values of the
+// specified labels in the metric. The hash is not based on the label names.
+func hashForLabels(metric clientmodel.Metric, labels clientmodel.LabelNames) uint64 {
+	// Sort labels to guarantee the same hash across numerous calls.
```

Comment:
hmm, is that a safe algorithm for this use case?

a{x="a",y="b"} will hash the same as a{x="b",y="a"}